### PR TITLE
HAL_ChibiOS: added support for H723 boards

### PR DESCRIPTION
**Pull request by tridge**
The H723 is a H7 with 1MByte single bank flash and a type 2 MCU config
the new board BotBloxSwitch is a ethernet switch peripheral acting as a PPP gateway

@andyp1per note the clock changes for the STM32H730. I aligned the no-crystal clocks with boards with a crystal and fixed the clock for FDCAN to be the required 80MHz

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H723xx.py [hunk 1]
```diff
@@ -0,0 +1,994 @@
...
+    # format is PIN : ADC1_CHAN
+    "PF11"  :   2,
+    "PA6"	:	3,
+    "PC4"	:	4,
+    "PB1"	:	5,
+    "PF12"  :   6,
+    "PA7"	:	7,
+    "PC5"	:	8,
+    "PB0"	:	9,
+	"PC0"	:	10,
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H723xx.py:87]: whitespace

File: libraries/AP_HAL_ChibiOS/hwdef/common/flash.c [hunk 2]
```diff
@@ -125,6 +125,7 @@ static const uint32_t flash_memmap[STM32_FLASH_NPAGES] = { KB(32), KB(32), KB(32
 #elif defined(STM32H7)
 #define STM32_FLASH_NPAGES  (BOARD_FLASH_SIZE / 128)
 #define STM32_FLASH_FIXED_PAGE_SIZE 128
+#define STM32_FLASH_NBANKS (BOARD_FLASH_SIZE/1024)
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/common/flash.c:128]: that is covered by the if defined(STM32H730xx) || defined(STM32H750xx)

**Review by andyp1per — APPROVED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H730xx.py [hunk 3]
```diff
@@ -69,7 +69,7 @@
 		('STM32_OSPICLK', 	200000000),
 		('STM32_SDMMC1CLK',  86666666),
 		('STM32_SPI45CLK',  100000000),
-		('STM32_FDCANCLK',   86666666),
+        ('STM32_FDCANCLK',   80000000),
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H730xx.py]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H723xx.py (hunk 1 above)
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H723xx.py]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H723xx.py [hunk 4]
```diff
@@ -0,0 +1,994 @@
...
+        (0x00000400,  63, 2), # ITCM (first 1k removed, to keep address 0 unused)
+        (0x30000000, 32, 8), # SRAM1, SRAM2
+        (0x38000000, 16, 1), # SRAM4
+    ],
+
+    'EXPECTED_CLOCK' : 400000000,
+    
+	'EXPECTED_CLOCKS' : [
+        ('STM32_SYS_CK',	400000000),
+		('STM32_OSPICLK', 	200000000),
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H723xx.py]: fixed

**Pull request merged**